metadata/src/prelude.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
```

Comment:
I think you can just add `extern crate alloc` to the top of lib.rs and no need to have it behind a feature flag. Then, you can just always point to the alloc and core versions of things directly as needed and avoid the need for this prelude entirely?

(Side note: polkadot-sdk has a similar prelude and also plans to remove it in favour of just pointing at alloc/core things instead where needed. I think it's generally a bit nicer to just avoid importing from std entirely like this rather than use features etc)

Reply:
Other side note: the annoying time when you cant just slap an `extern crate alloc` at the root of the library is when you're in a macro generating code. Then, you don't know if the library your macro is spitting code out into has used `extern crate alloc` or not, and then a nice approach is to pull imports from some known library (eg subxt macro would pull imports from subxt) to avoid needing an std feature flag or whatever)